#!/usr/bin/env python
PACKAGE = "fub_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("steering_lookahead_time",                double_t, 0, "time [s] to look ahead on the plan",                    1.0,      0,      5)
gen.add("steering_min_lookahead_dist",            double_t, 0, "min distance [m] to look ahead on the plan",            0.2,      0,     10)
gen.add("velocity_lookahead_time",                double_t, 0, "time [s] to look ahead on the plan",                    0,        0,      5)
gen.add("velocity_lookahead_dist_error_factor",   double_t, 0, 
    "factor for adjusting velocity to distance between closest point in space and lookahead",                           0.2,      0,      5)
gen.add("velocity_max_offset",                    double_t, 0, "maximum offset to velocity in plan [m/s].",             1.0,      0,     10)
gen.add("static_wanted_speed",           double_t, 0, "static wanted speed for debug 700 rpm=0.47 m/a",                 0.5,      0,    1.5)
gen.add("use_static_wanted_speed",       bool_t,   0, "whether to use the static speed",                                False)

exit(gen.generate(PACKAGE, "fub_controller", "ControllerMigPath"))
